brainiak/utils/fmrisim.py

```diff
@@ -539,56 +573,144 @@ def export_stimfunction(stimfunction,
-                     temporal_resolution=1000.0,
-                     ):
-    """Return a double gamma HRF
+def export_epoch_file(stimfunction,
+                      filename,
+                      tr_duration,
+                      temporal_resolution=1000.0
+                      ):
+    """ Output an epoch file, necessary for some inputs into brainiak
+
+    This takes in the time course of stimulus events and outputs the epoch
+    file used in Brainiak.
 
     Parameters
     ----------
-        stimfunction : list, bool
-            What is the time course of events to be modelled in this
-            experiment
 
-        tr_duration : float
+    stimfunction : list of timepoint by condition arrays
+        The stimulus function describing the time course of events. Each
+        list entry is from a different participant, each row is a different
+        timepoint, each column is a different condition
 
-        response_delay : float
-            How many seconds until the peak of the HRF
+    filename : str
+        The name of the three column text file to be output
 
-        undershoot_delay : float
-            How many seconds until the trough of the HRF
+    tr_duration : float
+        How long is each TR in seconds
 
-        response_dispersion : float
-            How wide is the rising peak dispersion
+    temporal_resolution : float
+        How many elements per second are you modeling with the
+        stimfunction?
 
-        undershoot_dispersion : float
-            How wide is the undershoot dispersion
+    """
 
-        response_scale : float
-             How big is the response relative to the peak
+    # Cycle through the participants, different entries in the list
+    epoch_file = [0] * len(stimfunction)
+    for participant_counter in list(range(0, len(stimfunction))):
 
-        undershoot_scale :float
-            How big is the undershoot relative to the trough
+        # What is the time course for the participant (binarized)
+        stimfunction_ppt = np.abs(stimfunction[participant_counter]) > 0
 
-        scale_function : bool
-            Do you want to scale the function to a range of 1
+        # Cycle through conditions
+        conditions = stimfunction_ppt.shape[1]
+        for condition_counter in list(range(0, conditions)):
 
-        temporal_resolution : float
-            How many elements per second are you modeling for the stim function
-    Returns
+            # Down sample the stim function
+            stride = tr_duration * temporal_resolution
+            stimfunction_temp = stimfunction_ppt[:, condition_counter]
+            stimfunction_temp = stimfunction_temp[::int(stride)]
+
+            if condition_counter == 0:
+                # Calculates the number of event onsets (max of all conditions)
+                epochs = int(np.max(np.sum((np.diff(stimfunction_ppt, 1,
+                                                    0) == 1), 0)) / 2)
```

Comment:
Fantastic find! I wasn't aware of this bug because all of my test cases were with zeros and ones. I have now simplified this to be agnostic to the weight value and when the change occurs. However, it still has some limitations (if there was no pause between epochs or the weights changed dynamically within epochs)